shell/browser/notifications/mac/cocoa_notification.h

```diff
@@ -35,10 +37,14 @@ class CocoaNotification : public Notification {
 
  private:
   void LogAction(const char* action);
+  // SAP-14036 upgrade for persistent notifications support
+  bool is_persistent_;
+  bool allow_destroy_;
```

Comment:
Even persistent notifications can be delivered as banners in case of support for requireInteraction
see logic here `is_persistent_ = options.is_persistent && options.require_interaction;`
allow_destroy is used only for clearing notifications after app is down. 
If into few words the purpose of these flags are different - one used to define type of delivery Alert/Banner and second for clean up already delivered notifications after app down.